lightningd/lightningd.c

```diff
@@ -965,6 +965,7 @@ int main(int argc, char *argv[])
 
 	/*~ Handle options and config. */
 	handle_opts(ld, argc, argv);
+	ld->portnum = DEFAULT_PORT + chainparams->port_add;
```

Comment:
```suggestion
      if (ld->portnum == DEFAULT_PORT )
	    ld->portnum = DEFAULT_PORT + chainparams->port_add;
```

Reply:
The condition looks nice, but in practice it is not needed. I have verified locally that command-line takes precedence still. No condition needed.

Reply:
Or is [the failed test](https://github.com/ElementsProject/lightning/runs/4092052054?check_suite_focus=true) caused by the missing condition?

Reply:
To me the failed test looks like any other randomly failed test (since 2018) without reason. I am happy the `lightningd` is not failing like that in the wild and is actually very stable and reliable. But by-standers can be mislead when they see such failing tests.

Reply:
It only needs more feed-back now :) so the condition is not important.